kratos/tests/cpp_tests/utilities/test_auxiliar_model_part_utilities.cpp

```diff
@@ -0,0 +1,558 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:   BSD License
+//      Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "testing/testing.h"
+#include "includes/model_part.h"
+#include "utilities/auxiliar_model_part_utilities.h"
+
+// Utilities
+#include "utilities/cpp_tests_utilities.h"
+
+namespace Kratos {
+namespace Testing {
+
+/******************************************************************************************/
+/* Helper Functions */
+/******************************************************************************************/
+
+void Initialise(ModelPart& this_model_part)
+{
+    this_model_part.AddNodalSolutionStepVariable(DISPLACEMENT);
+
+    CppTestsUtilities::Create2DGeometry(this_model_part, "Element2D3N", true, true);
+}
+
+std::vector<double> ComputationGetData(const DataLocation DataLoc, ModelPart& this_model_part, int dim)
+{
+    //Assign random values to Displacement of each node and also save it in test_values for comparision
+    int i=0; //To create various values
+    double disp_test_value = 1.26;
+    std::vector<double> test_values;
+
+    switch (DataLoc)
+        {
+            case (DataLocation::NodeHistorical):{
+                test_values.resize(this_model_part.NumberOfNodes() * dim);
+                for (auto& i_node : this_model_part.Nodes())
+                {
+                    for(int j=0; j<dim; j++)
+		            {
+                        i_node.FastGetSolutionStepValue(DISPLACEMENT)[j] = static_cast<double>(disp_test_value * (i+j));
+                        test_values[i] = (disp_test_value * (i+j));
+                        i++;
+                    }
+                }
+                break;
+            }
+            case (DataLocation::NodeNonHistorical):{
+                test_values.resize(this_model_part.NumberOfNodes() * dim);
+                for (auto& i_node : this_model_part.Nodes())
+                {
+                    //array_1d<double, 3> test_vals; // can be done like this
+		            for(int j=0; j<dim; j++)
+		            {
+                        //i_node.SetValue(DISPLACEMENT, test_vals);
+                        i_node.GetValue(DISPLACEMENT)[j] = static_cast<double>(disp_test_value * (i+j));
```

Comment:
why do you need the static_casts here?

Reply:
Not needed I guess .. Changed and committed .